content/posts/2025-10-30_mastodon_4_5_for_devs/index.md

```diff
@@ -0,0 +1,65 @@
+description: "Here's what developers need to know about the upcoming Mastodon 4.5 release."
+date: 2025-10-30
+slug: mastodon-4-5-for-devs
+authors:
+ - mastodon
+tags:
+ - mastodon
+ - engineering
+categories:
+ - Engineering
+resources:
+- name: hero
+  src: hero.png
+draft: false
+---
+
+Mastodon 4.5 is almost ready. We just released the first (and hopefully only) [release candidate](https://github.com/mastodon/mastodon/releases/tag/v4.5.0-rc.1) and the final release should be coming along very soon. This is once again a good time to highlight some of the new features and changes that are important to anyone building apps or integrations.
+
+As a reminder, we also share a monthly engineering update (our [Trunk and Tidbits](https://blog.joinmastodon.org/categories/trunk-and-tidbits/) series) and if you’ve been following those, some of these changes may already be familiar. You’ll see below that there are a lot of useful new additions that have arrived during this development cycle.
+
+We’ve divided this post into two broad sections: updates that are likely to be most relevant for developers of client applications, and updates that are more applicable to the broader Fediverse, other integrations, or contributors to Mastodon itself. Let’s dive in.
+
+## Client apps
+
+### Quote Posts
+
+Mastodon 4.4 released with the capability to verify and display quote posts as defined by our specification, [FEP-044f](https://codeberg.org/fediverse/fep/src/branch/main/fep/044f/fep-044f.md).
+
+Mastodon 4.4 introduced two new API entities to represent quotes, [Quote](https://docs.joinmastodon.org/entities/Quote/) and [ShallowQuote](https://docs.joinmastodon.org/entities/ShallowQuote/). Mastodon 4.5 improves those by adding new states: `blocked_account`, `blocked_domain`, `muted_account`. These are cases where the quote is authorized but should not be displayed by default to the user currently viewing the post. In Mastodon 4.4 we just returned a state of `unauthorized` in those cases, but we decided that applications should be able to offer a way to see the quoted posts anyway. See the documentation of the entities above and [this PR](https://github.com/mastodon/mastodon/pull/36619).
+
+The biggest change compared to 4.4 is of course that we now allow authoring of quote posts. In the REST API this can be as simple as adding a `quoted_status_id` attribute when creating a new status. But clients should at least make sure that quoting the status in question is not disallowed. To help you get acquainted with all the relevant changes to the API, we added an ["Implementing quote posts" guide](https://docs.joinmastodon.org/client/quotes/).
+
+### Fetch all replies / Async Refreshes
+
+Mastodon 4.4 introduced a new feature to fetch missing replies to a remote status. This was not enabled by default and had no immediate effect in any user interfaces. Mastodon 4.5 enables the feature by default and also includes some user facing changes in the web UI around this.
```

Comment:
“Mastodon 4.5 enables the feature by default” sounds like it can be disabled, which is not the case.